### PR TITLE
slackcat: init at 1.6

### PR DESCRIPTION
**Pull request by nasadorian**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nasadorian** [comment]
@GrahamcOfBorg How can I get this PR merged?

File: pkgs/tools/networking/slackcat/default.nix [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+buildGoPackage rec {
+  pname = "slackcat";
+  version = "1.6";
+  goPackagePath = "github.com/bcicen/slackcat";
+  src = fetchFromGitHub {
+    owner = "bcicen";
+    repo = "slackcat";
+    rev = "v$version";
+    sha256 = "1ha9b2dffpxqs159ldqdh1aibn3316xab19vdn2yjgk8r508wm31";
+  };
```
Comment (dali99) [inline pkgs/tools/networking/slackcat/default.nix:12]:
```suggestion
  src = fetchFromGitHub {
    owner = "bcicen";
    repo = "${pname}";
    rev = "v${version}";
    sha256 = "1ha9b2dffpxqs159ldqdh1aibn3316xab19vdn2yjgk8r508wm31";
  };
```

**Review by D4ndellion — CHANGES_REQUESTED** [review]
All packages need meta information

**Comment by nasadorian** [comment]
@dali99 Thanks for your review. Do I need to add myself as a maintainer or the package developer himself?

**Comment by dali99** [comment]
> @dali99 Thanks for your review. Do I need to add myself as a maintainer or the package developer themself?

You add yourself

For more information you can read
https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md and the linked guides in it :smiley: 

In your case you should have two commits
One adding you to the maintainers list
```
maintainers: add nasadorian
```
With the relevant information to `maintainers/maintainer-list.nix `

The other commit should be
```
slackcat: init at 1.6
```
adding the package itself.

There should be no more commits (so you need to force push your branch)

**Comment by dali99** [comment]
You seem to have based your PR on a branch from May in 2019

The `maintainers/maintainer-list.nix` has changed since then, mainly including a `githubId`.

```
nix
    handle = {
      # Required
      name = "Your name";
      email = "address@example.org";
      # Optional
      github = "GithubUsername";
      githubId = your-github-id;
      keys = [{
        longkeyid = "rsa2048/0x0123456789ABCDEF";
        fingerprint = "AAAA BBBB CCCC DDDD EEEE  FFFF 0000 1111 2222 3333";
      }];
    };
```

Just commenting so that you'll be aware of this. I'd rebase on current master and add at least `githubId`

**Review by D4ndellion — APPROVED** [review]
Built sucessfully!

:tada:

**Comment by nasadorian** [comment]
@dali99 Is there a way to kick the build? I'm not sure what happened here.

**Comment by dali99** [comment]
ah yes we should ping someone
@jonringer ?

**Comment by pbogdan** [comment]
@GrahamcOfBorg eval

**Comment by nasadorian** [comment]
Boom passes! Thanks @dali99 @pbogdan

**Comment by nasadorian** [comment]
Hi guys, how can I get this PR merged? @dali99 @pbogdan

File: pkgs/tools/networking/slackcat/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    rev = "v${version}";
+    sha256 = "1ha9b2dffpxqs159ldqdh1aibn3316xab19vdn2yjgk8r508wm31";
+  };
+  goDeps = ./deps.nix;
+  meta = with stdenv.lib; {
+    description = "CLI utility to post files and command output to slack";
+    homepage = "http://slackcat.chat/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ nasadorian ];
+  };
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/slackcat/default.nix:19]:
```suggestion
  version = "1.6";

  src = fetchFromGitHub {
    owner = "bcicen";
    repo = "${pname}";
    rev = "v${version}";
    sha256 = "1ha9b2dffpxqs159ldqdh1aibn3316xab19vdn2yjgk8r508wm31";
  };

  goPackagePath = "github.com/bcicen/slackcat";

  meta = with stdenv.lib; {
    description = "CLI utility to post files and command output to slack";
    homepage = "http://slackcat.chat/";
    license = licenses.mit;
    maintainers = with maintainers; [ nasadorian ];
  };
```

buildGoPackage is legacy and nowday we prefer buildGoModule. Please switch to that.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/slackcat/default.nix (hunk 2 above)
Comment (nasadorian) [reply pkgs/tools/networking/slackcat/default.nix]: Thanks @SuperSandro2000 committed

File: pkgs/tools/networking/slackcat/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+buildGoPackage rec {
+  pname = "slackcat";
+  version = "1.6";
+
+  src = fetchFromGitHub {
+    owner = "bcicen";
+    repo = "${pname}";
+    rev = "v${version}";
+    sha256 = "1ha9b2dffpxqs159ldqdh1aibn3316xab19vdn2yjgk8r508wm31";
+  };
```
Comment (jonringer) [inline pkgs/tools/networking/slackcat/default.nix:12]: I noticed they have a go.mod https://github.com/bcicen/slackcat/blob/0b70f6e0ab9c41e4782bfbe74e934ddb49d520b8/go.mod, do you mind trying to do a `buildGoModule` instead? package is deprecated

File: pkgs/tools/networking/slackcat/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  pname = "slackcat";
+  version = "1.6";
+
+  src = fetchFromGitHub {
+    owner = "bcicen";
+    repo = "${pname}";
```
Comment (jonringer) [inline pkgs/tools/networking/slackcat/default.nix:9]:
```suggestion
    repo = pname;
```

**Comment by jonringer** [comment]
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

 `git rebase -i` is a powerful command which achieves this, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

File: pkgs/tools/networking/slackcat/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/slackcat/default.nix:3]: Please use buildGoModule which does not require use to use a deps.nix.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**